shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -104,7 +104,16 @@ final class TsTypegenCodeBuilder {
             // flip polarity for input type of function
             // lhs translates to the complete argument list
             // method definition only affects source code representation of top level function
-            val lhsTypeSource = toTsType(lhs, true)(typegenCtx, Some(false))
+            val lhsTypeSource = lhs match {
+              // tuple are handled specially because
+              // they create their own arg names
+              case Tuple(fields) => toTsType(lhs, true)(typegenCtx, Some(false))
+              // there are other types which might hide tuple arguments
+              // like recursive types, AppliedTypes or bounded types
+              // their tuple types can be uniformly converted
+              // using a spread operator
+              case _ => (SourceCode("...arg: ") ++ toTsType(lhs, true)(typegenCtx, Some(false))).parenthesized
```

Comment:
This works, but for consistency, shouldn't we generate a new "arg" name using the scope methods?

Reply:
Yes made the change.